app/src/main/java/com/example/lapnguyen/weather_3/screen/mainscreen/MainPresenter.java

```diff
@@ -0,0 +1,64 @@
+package com.example.lapnguyen.weather_3.screen.mainscreen;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.location.Criteria;
+import android.location.Location;
+import android.location.LocationManager;
+import android.support.v4.app.ActivityCompat;
+import android.util.Log;
+import com.example.lapnguyen.weather_3.data.model.Weather;
+import com.example.lapnguyen.weather_3.data.source.MainRepository;
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.tasks.OnSuccessListener;
```

Comment:
ERROR: Unused import: com.google.android.gms.tasks.OnSuccessListener. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports